topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -406,12 +413,62 @@ private TopicStore createTopicStore(Zk zk, Config config) {
     }
 
     private void setSaslConfigs(Properties kafkaClientProps) {
-        String saslMechanism;
-        String jaasConfig;
+        String configSaslMechanism = config.get(Config.SASL_MECHANISM);
+        
+        switch (configSaslMechanism) {
+            case SASL_TYPE_PLAIN:
+            case SASL_TYPE_SCRAM_SHA_256:
+            case SASL_TYPE_SCRAM_SHA_512:
+                setStandardSaslConfigs(kafkaClientProps);
+                break;
+            case "custom":
+                setCustomSaslConfigs(kafkaClientProps);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid SASL_MECHANISM: '" + configSaslMechanism + "'");
+        }
+    }
+
+    private void setCustomSaslConfigs(Properties kafkaClientProps) {
+        String customPropsString = config.get(Config.SASL_CUSTOM_CONFIG);
+
+        if (customPropsString.isEmpty()) {
+            throw new InvalidConfigurationException("Custom SASL config properties are not set");
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            Map<String, String> customProperties = objectMapper.readValue(customPropsString, STRING_HASH_MAP_TYPE_REFERENCE);
+
+            if (customProperties.isEmpty()) {
+                throw new InvalidConfigurationException("SASL custom config properties empty");
+            }
+
+            for (var entry : customProperties.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+
+                if (key == null || key.isBlank() || !key.startsWith("sasl.")) {
+                    throw new InvalidConfigurationException("SASL custom config properties not SASL properties. customProperty: '" + key + "' = '" + value + "'");
+                }
+
+                kafkaClientProps.setProperty(key, value);
+            }
+        } catch (JsonProcessingException e) {
+            throw new InvalidConfigurationException("SASL custom config properties deserialize failed. customProperties: '" + customPropsString + "'");
```

Comment:
We should include any message from `e` (it might include line number and/or say what's wrong, which would help the user identify the problem).

```suggestion
            throw new InvalidConfigurationException("Env var " + Config.SASL_CUSTOM_CONFIG + " could not be parsed as JSON + (e.getMessage() != null ? ": " + e.getMessage() : "."));
```